controllers/nstemplateset/nstemplateset_controller_test.go

```diff
@@ -1766,45 +1786,78 @@ func getTemplateTiers(decoder runtime.Decoder) ([]runtime.Object, error) {
 					default:
 						tmplContent = test.CreateTemplate(test.WithObjects(ns, rb, role), test.WithParams(username))
 					}
+
 				}
-				tmplContent = strings.ReplaceAll(tmplContent, "nsType", typeName)
-				tmpl := templatev1.Template{}
-				_, _, err := decoder.Decode([]byte(tmplContent), nil, &tmpl)
+				tierTmpl, err := createTierTemplate(decoder, tmplContent, tierName, typeName, revision)
 				if err != nil {
 					return nil, err
 				}
-				tierTmpl := &toolchainv1alpha1.TierTemplate{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      strings.ToLower(fmt.Sprintf("%s-%s-%s", tierName, typeName, revision)),
-						Namespace: test.HostOperatorNs,
-					},
-					Spec: toolchainv1alpha1.TierTemplateSpec{
-						TierName: tierName,
-						Type:     typeName,
-						Revision: revision,
-						Template: tmpl,
-					},
-				}
 				tierTemplates = append(tierTemplates, tierTmpl)
 			}
 		}
 	}
+	// ---- appstudio templates ----
+	// namespace template
+	tierTmpl, err := createTierTemplate(decoder,
+		test.CreateTemplate(test.WithObjects(ns), test.WithParams(namespace, username)),
+		"appstudio", "appstudio", "abcde11")
+	if err != nil {
+		return nil, err
+	}
+	tierTemplates = append(tierTemplates, tierTmpl)
+	// space roles templates
+	tierTmpl, err = createTierTemplate(decoder,
+		test.CreateTemplate(test.WithObjects(spaceAdmin, spaceAdminRb), test.WithParams(namespace, username)),
+		"admin", "appstudio", "abcde11")
+	if err != nil {
+		return nil, err
+	}
+	tierTemplates = append(tierTemplates, tierTmpl)
+	tierTmpl, err = createTierTemplate(decoder,
+		test.CreateTemplate(test.WithObjects(spaceViewer, spaceViewerRb), test.WithParams(namespace, username)),
+		"viewer", "appstudio", "abcde11")
+	if err != nil {
+		return nil, err
+	}
```

Comment:
as we discussed in the call - the SpaceRoles are not `appstudio` specific, but they will be used in all tiers. 
I would propose dropping this part and incorporating the SpaceRoles into the already existing tier setup - otherwise, it could bring additional confusion why we need a special `appstudio` tier for that.

Reply:
done in 4c78c4e. I also refactored the `prepareTemplateTiers` func so that it's visually easier to see what templates are generated